components/Optional/index.js

```diff
@@ -0,0 +1,16 @@
+import PropTypes from 'prop-types';
+import { useBlockEditContext } from '@wordpress/block-editor';
+
+export const Optional = ({ value, children }) => {
+	const { isSelected } = useBlockEditContext();
+	return !!value && isSelected && children;
```

Comment:
Sorry I missed this in the first round. This actually is not working as expected. 

The children should always get rendered if the block is selected. Only when the block is not selected and the value is falsey should the component be hidden. 

```suggestion
	return ( isSelected || !!value ) && children;
```

Reply:
I built the component following the example in the issue which had this logic along the lines - `!! title && isSelected && children` 
Should we update the issue description?